js/test/wallet/createTransaction.test.ts

```diff
@@ -0,0 +1,436 @@
+import {
+    Connection,
+    Keypair,
+    LAMPORTS_PER_SOL,
+    PublicKey,
+    SystemProgram,
+    Transaction,
+    TransactionInstruction,
+} from '@solana/web3.js';
+import BigNumber from 'bignumber.js';
+import { createTransaction } from '../../src/wallet/createTransaction';
+import { getMint, getAccount, getAssociatedTokenAddress } from '@solana/spl-token';
+
+jest.mock('@solana/spl-token', () => ({
+    getMint: jest.fn(),
+    getAccount: jest.fn(),
+    getAssociatedTokenAddress: jest.fn(),
+}));
+
+const mockGetMint = getMint as jest.Mock;
+const mockGetAccount = getAccount as jest.Mock;
+const mockGetAssociatedTokenAddress = getAssociatedTokenAddress as jest.Mock;
+
+describe('createTransaction', () => {
+    let connection: Connection;
+    let wallet: Keypair;
+    const recipientPublicKey = new PublicKey('mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN');
+    const accountOwnedByWallet = Keypair.generate();
+    const programAccount = Keypair.generate();
+
+    beforeAll(async () => {
+        wallet = Keypair.generate();
+
+        connection = {
+            getAccountInfo: (publicKey: PublicKey) => {
+                const accounts = {
+                    [wallet.publicKey.toBase58()]: {
+                        executable: false,
+                        owner: SystemProgram.programId,
+                        lamports: LAMPORTS_PER_SOL / 100,
+                    },
+                    [recipientPublicKey.toBase58()]: {
+                        executable: false,
+                        owner: SystemProgram.programId,
+                        lamports: LAMPORTS_PER_SOL / 100,
+                    },
+                    [SystemProgram.programId.toBase58()]: {
+                        executable: true,
+                        owner: SystemProgram.programId,
+                        lamports: LAMPORTS_PER_SOL / 100,
+                    },
+                    [accountOwnedByWallet.publicKey.toBase58()]: {
+                        executable: false,
+                        owner: wallet.publicKey,
+                        lamports: LAMPORTS_PER_SOL / 100,
+                    },
+                    [programAccount.publicKey.toBase58()]: {
+                        executable: true,
+                        owner: SystemProgram.programId,
+                        lamports: LAMPORTS_PER_SOL / 100,
+                    },
+                };
```

Comment:
This gets created every time we call `getAccountInfo`. Shouldn't we just create it once?